ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/error/PostSSOErrorTest.kt

```diff
@@ -93,9 +95,16 @@ class PostSSOErrorTest : StringSpec() {
                     val encodedRequest = signAndEncodePostRequestToString(authnRequest,
                         EXAMPLE_RELAY_STATE)
                     val response = sendPostAuthnRequest(encodedRequest)
+                    BindingVerifier.verifyHttpStatusCode(response.statusCode)
 
-                    if (!isLenient || !BindingVerifier.isErrorHttpStatusCode(response.statusCode)) {
-                        val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
+                    val finalHttpResponse =
+                            TestCommon.getImplementation(IdpSSOResponder::class)
+                                    .getResponseForPostRequest(response)
```

Comment:
I think I'm okay with this change until we run into an IdP that acts differently, but I think our IdP needing more interaction before validating the Subject is implementation-dependent.

Reply:
I agree that this is implementation dependent. I also agree that in order to fix this issue, a separate effort should be made to look into:
1. How many of our tests are actually implementation dependent
2. The best solution to solve this implementation dependence.

Implementation dependent example: IdP1 could take the AuthnRequest Subject and check it against its user database, sending an error response without having to log any user in. Whereas IdP2 (lookin' at you DDF) may log the user in, then compare that user to the Subject sent in the AuthnRequest.